[compile error] C++20 removed error std::result_of used in <pcl/common/common.h>

**Describe the error**
Using the VCPKG port of PCL with Version: 1.12.0

C++20 removed the `std::result_of` call (see https://en.cppreference.com/w/cpp/types/result_of) which results in compile errors when including `<pcl/common/common.h>`.

```c++
template<typename IteratorT, typename Functor> inline auto
computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept -> typename std::result_of<Functor(decltype(*begin))>::type
```

The exact compile error is as follows:
`pcl/common/common.h(285): error C2653: 'result_of': is not a class or namespace name`

**To Reproduce**
Install PCL via vcpkg and in a MSVC project using /std:c++20 include `<pcl/common/common.h>`.

or

Just enable a C++20 compliant compiler when including `<pcl/common/common.h>`

**Your Environment (please complete the following information):**

 - OS: Windows 10
 - Compiler: Visual Studio 2019 16.11.7
 - PCL Version 1.12.0
 - PCL Type: Compiled via VCPKG with /arch:AVX


**Possible Solution**
When detecting a compiler with C++17 or greater use `std::invoke_result` instead (https://en.cppreference.com/w/cpp/types/result_of).

--- a/common/include/pcl/common/common.h
+++ b/common/include/pcl/common/common.h
@@ -284,7 +284,7 @@ namespace pcl
   template<typename IteratorT, typename Functor> inline auto
   computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept ->
   #if __cpp_lib_is_invocable
-  std::invoke_result_t<Functor(decltype(*begin))>
+  std::invoke_result_t<Functor, decltype(*begin)>
   #else
   typename std::result_of<Functor(decltype(*begin))>::type
   #endif

--- a/common/include/pcl/common/common.h
+++ b/common/include/pcl/common/common.h
@@ -282,7 +282,12 @@ namespace pcl
     * \ingroup common
     */
   template<typename IteratorT, typename Functor> inline auto
-  computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept -> typename std::result_of<Functor(decltype(*begin))>::type
+  computeMedian (IteratorT begin, IteratorT end, Functor f) noexcept ->
+  #if __cpp_lib_is_invocable
+  std::invoke_result_t<Functor(decltype(*begin))>
+  #else
+  typename std::result_of<Functor(decltype(*begin))>::type
+  #endif
   {
     const std::size_t size = std::distance(begin, end);
     const std::size_t mid = size/2;

